assert/assertions.go

```diff
@@ -918,11 +932,37 @@ func ElementsMatch(t TestingT, listA, listB interface{}, msgAndArgs ...interface
 			}
 		}
 		if !found {
-			return Fail(t, fmt.Sprintf("element %s appears more times in %s than in %s", element, aValue, bValue), msgAndArgs...)
+			extraA = append(extraA, element)
 		}
 	}
 
-	return true
+	for j := 0; j < bLen; j++ {
+		if !visited[j] {
```

Comment:
Would it be neater if we changed this to a guard clause? e.g.
```
for .... {
    if visited[j] {
        continue
    }
    extraB = append(...)
}
```

Reply:
Updated.